tools/circt-reduce/circt-reduce.cpp

```diff
@@ -104,6 +104,12 @@ static cl::opt<bool> verbose("v", cl::init(true),
                              cl::desc("Print reduction progress to stderr"),
                              cl::cat(mainCategory));
 
+cl::opt<int64_t> maxGreedyRewrites(
+    "max-greedy-rewrites", cl::init(-1),
+    cl::desc("Maximum number of rewrites GreedyPatternRewriteDriver may "
+             "apply (negative value keeps the default)"),
+    cl::cat(mainCategory));
```

Comment:
Nit: I'd just use "num" instead of "greedy" to match the `circt-opt` help text.

```suggestion
cl::opt<int64_t> maxNumRewrites(
    "max-num-rewrites", cl::init(-1),
    cl::desc("Maximum number of rewrites GreedyPatternRewriteDriver may "
             "apply (negative value keeps the default)"),
    cl::cat(mainCategory));
```